plugins/org.locationtech.udig.tool.edit/src/org/locationtech/udig/tools/edit/behaviour/accept/AcceptChangesBehaviour.java

```diff
@@ -265,19 +267,20 @@ public UndoableMapCommand getCommand(EditToolHandler handler) {
                     // commands.add(newFeatureCommand);
                     // }
                 } else {
-                    String idRef = editGeom.getFeatureIDRef().get();
-                    if (idRef != null) {
+                    if (featureIdReference != null) {
                         // not creating it so don't need to set it.
-                        UndoableMapCommand setGeometryCommand = new SetGeometryCommand(idRef,
+                        UndoableMapCommand setGeometryCommand = new SetGeometryCommand(featureIdReference,
                                 new StaticBlockingProvider<>(layer), SetGeometryCommand.DEFAULT,
                                 geom);
                         commands.add(setGeometryCommand);
                     }
                 }
             } else {
-                commands.add(new CreateNewOrSelectExitingFeatureCommand(
-                        editGeom.getFeatureIDRef().get(), layer, geom));
-
+                if (featureIdReference != null) {
+                    CreateNewOrSelectExitingFeatureCommand featureCommand = new CreateNewOrSelectExitingFeatureCommand(
```

Comment:
@fgdrf may I ask if I just found another typo here? As far as I can see it should be "CreateNewOrSelectExistingFeatureCommand" right?

If yes, I will rename the class after this PR is merged.

Reply:
Good catch, we can adress it in another changest ;)